built_value_generator/lib/src/serializer_source_class.dart

```diff
@@ -515,35 +515,87 @@ class $serializerImplName implements PrimitiveSerializer<$genericName> {
       final cast = field.generateCast(compilationUnit, _genericBoundsAsMap);
       // If cast exists and is not nullable.
       var maybeNotNull = !field.isNullable && cast.isNotEmpty ? notNull : '';
+      var oldWireNamesCaseDeclarations =
+          field.oldWireNames.map((name) => "case '$name':").join('\n');
       if (field.builderFieldUsesNestedBuilder) {
         if (field.builderFieldAutoCreatesNestedBuilder || hasBuilder) {
-          return '''
-case '${escapeString(field.wireName)}':
+          final oldWireNamesCases = '''
+$oldWireNamesCaseDeclarations
+  if(\$haveSet${field.capitalizedName}) break;
   result.${field.name}.replace(serializers.deserialize(
       value, specifiedType: $fullType)$notNull $cast);
+  \$haveSet${field.capitalizedName} = true;
   break;
 ''';
+          final oldWireNameSetter = field.oldWireNames.isEmpty
+              ? ''
+              : '\$haveSet${field.capitalizedName} = true;';
+          final fundamentalFieldCase = '''
+case '${escapeString(field.wireName)}':
+  result.${field.name}.replace(serializers.deserialize(
+      value, specifiedType: $fullType)$notNull $cast); $oldWireNameSetter
```

Comment:
Here's another example of putting a generated snippet on the same line to reduce diffs in existing generated built_value code

Reply:
Looks good, thanks.